From 9e19461dff24bd7ec4a566677f5d8678274987c3 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 12 Feb 2019 22:21:05 +0100
Subject: GCC: fix canvas2d rendering context type conversion.

There was a creation of an uint8_t[] array using ValueOrDie of a
data length (a checked numeric of size_t). But GCC cannot properly
extract the type, so we need an explicit cast to size_t.

diff --git a/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc b/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
index d9fa696c9a9d..e0399187293c 100644
--- a/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
+++ b/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
@@ -1774,7 +1774,7 @@ void BaseRenderingContext2D::putImageData(ImageData* data,
     if (!data_length.IsValid())
       return;
     std::unique_ptr<uint8_t[]> converted_pixels(
-        new uint8_t[data_length.ValueOrDie()]);
+        new uint8_t[static_cast<size_t>(data_length.ValueOrDie())]);
     if (data->ImageDataInCanvasColorSettings(
             ColorParams().ColorSpace(), PixelFormat(), converted_pixels.get(),
             kRGBAColorType)) {
-- 
2.17.1

